### ISSUE TITLE
[BUG] Cryptic upgrade message when overwritten core module tids

### ISSUE THREAD

**Issue by twMat** [Jermolene/TiddlyWiki5#7403]
### Describe the bug

The following message appears when using  https://tiddlywiki.com/upgrade to upgrade to 5.2.7 (from an old 5.2.3 version, but I assume the message appears irrespective of old version)

![image](https://user-images.githubusercontent.com/4534056/231981976-842010d6-b569-4da6-9f06-1e0c26698d8b.png)

This message is problematic because it:

1. does not convey what the user should do
2. is not clear what tiddler it actually refers to
3. The message is probably misleading in that it states "**a** tiddler" when it probably can concern an arbitrary number of tiddlers. If the message actually prompts a user to *take action* then phrasing it as "**a** tiddler" risks making a user take "one action" when it might really require multiple actions.)

A more experienced user knows that the alert refers to the list of tiddler titles below the message, but it is still not clear which tiddler(s) the alert refers to as a problem. The only *description* of the problematic tiddler(s) in the alert is that it is a "core module tiddler". Scrolling the list **only** shows the following described as a "Core module tiddler" (i.e the description in the right hand column):

![image](https://user-images.githubusercontent.com/4534056/231989735-71021368-d433-44ed-a167-be5a7b609088.png)

...but this is not really what the message refers to (...right?) Further misdirection comes when noticing that the tiddler title just above this also is prefixed `$:/core/modules` ...implying that it is, indeed, a core module tiddler but, "evidently, not the one that the alert wants you to notice":

![image](https://user-images.githubusercontent.com/4534056/231990223-cf73def2-a207-4a69-bba3-89cd7e81fe20.png)

I (an experienced user) have thus far assumed that these alerts refer to **ALL overwritten shadow tiddlers** and that these are therefore highlighted in *red* in the list, like the following:

![image](https://user-images.githubusercontent.com/4534056/231992740-8fed1e44-22f8-439b-9cc7-60973b425c57.png)

...but, thinking about it, I'm actually not sure anymore. (Would anyone be so kind to inform me, please? This is not my primary reason for posting though.)

### Expected behavior

The message should direct the user to which titles are actually concerned, presumably by referring to the list of tiddlers below it and clarifying how to identify which tiddler(s!) are referred.

Alternatively, the message could itself list the concerned tiddlers - perhaps by integrating the message into the top of the list so to present a separate list before the complete list below it.

### To Reproduce

Update an old TW that contains overwritten core tids (....at least this is how I *think* you would reproduce it, since I'm assuming the alert really refers to these... but just maybe I misunderstand...)

### TiddlyWiki Configuration

- Version 5.2.3 into 5.2.7
- From local file, using the standard online upgrader
- Plugins installed: a large number of plugins and tweaks are installed in this particular wiki

Win10, Brave, Acer laptop

**Comment by pmario** [comment]
>Update an old TW that contains overwritten core tids (....at least this is how I think you would reproduce it, since I'm assuming the alert really refers to these... but just maybe I misunderstand...)

You are right. If I change eg the `startup/widows.js` tiddler and upgrade again, the message can be reproduced. 

So... yes ...  there should be a better info

**Comment by pmario** [comment]
IMO the **To Reproduce** info should be at the very top of the issue. .. I did read that info, after I did post my first reply. ...

**Comment by Jermolene** [comment]
Thank you @twMat I think there are a several aspects to this issue:

* The use of an alert message makes it harder to associate it with what is shown in the $:/Import tiddler
* The alert message might be applied too broadly. It is  issued when attempting to import a tiddler whose title starts with `$:/core/modules/` and there is already a shadow tiddler in the wiki with the same title (in your example I believe that the message is triggered by the override of `$:/core/modules/startup/windows.js`)

**Comment by twMat** [comment]
>You are right. If I change eg the `startup/widows.js` tiddler and upgrade again, the message can be reproduced.

Thanks @pmario  - Aha, so the problematic tiddler was really (and only)  `startup/widows.js` after all!

@Jermolene and @pmario : Which issues trigger alerts in $:/Import ? It's not only overwritten core modules, is it?

Alerts would not be needed if tiddlers requiring attention could be listed separately before the full Import table such a separate list can itself be designed to call for user attention. If such a list is not possible, another idea would be to simplify the alert popup into a very generic `"Attention! Please inspect the highlighted tiddler(s) in the list"` and then, in the list and next to the relevant titles, include the problem descriptions and what to do about it.

By the way; Getting an explicit *alert* also signals something different than the *red highlight* in terms of how severe the "problem" is. Is that the case? I doubt it considering how the red highlights, i.e overwritten core tids (...and more?) can also be super critical tiddlers (...right?). Maybe the following is exactly what the alerts refer to, in contrast to the red table rows, but if not: There could be a distinction between system tiddlers that are *critical* vs *non-critical* **for the TW to function at all**. And the Import list uses two different signals to convey the severity of the overwrite. (Also advanced plugin creators could be informed about how to mark out tiddlers to classify them as system critical).

One idea to signal such "degree of severity" could be to, in the Import table, use red cell *borders* for "smaller" alerts and red *backgrounds* for "critical" ones. Pretty neat, IMO ;-)

**Comment by pmario** [comment]
The upgrade mechanism only has 1 indicator, that causes an alert. As Jeremy wrote it's the `$:/core/modules/` prefix. 

It should be possible to mark those tiddlers with a red background and change the text in the alert to "You are about ,,, Check all red fields in the import table." ... or something similar. 

The problem is, that the upgrader has no interface to check what's different. The Import mechanism has a possibility to show "diffs" ... The upgrader does not. 

So the only way I can think of is, to "unselect" `$:/core/module/` tiddlers and mark them red. ... At the moment they are "selected" and _not highlighted_ .. It's just the alert, which does not really help the user.
